packages/connectivity/src/scp-cf/destination/destination-cache.spec.ts

```diff
@@ -317,6 +293,106 @@ describe('destination cache', () => {
     });
   });
 
+  describe('caching options', () => {
+    beforeEach(() => {
+      mockServiceBindings();
+      mockServiceToken();
+      mockVerifyJwt();
+    });
+
+    const destName = destinationOne.name!;
+
+    it('disables the cache by default', async () => {
+      destinationCache.cacheRetrievedDestination(
+        { user_id: 'user', zid: 'tenant' },
+        destinationOne,
+        IsolationStrategy.Tenant
+      );
+      await expect(
+        getDestination({ destinationName: destName })
+      ).rejects.toThrowError(/Failed to fetch \w+ destinations./);
+    });
+
+    it('uses cache with isolation strategy Tenant if no JWT is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(providerServiceToken),
+        destinationOne,
+        IsolationStrategy.Tenant
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        useCache: true
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+
+    it('uses cache with isolation strategy TenantUser if JWT is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(subscriberUserJwt),
+        destinationOne,
+        IsolationStrategy.Tenant_User
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        useCache: true,
+        jwt: subscriberUserJwt,
+        iasToXsuaaTokenExchange: false
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+
+    it('enables cache if isolation strategy Tenant is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(providerServiceToken),
+        destinationOne,
+        IsolationStrategy.Tenant
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        isolationStrategy: IsolationStrategy.Tenant,
+        iasToXsuaaTokenExchange: false
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+
+    it('enables cache if isolation strategy TenantUser is provided', async () => {
+      destinationCache.cacheRetrievedDestination(
+        decodeJwt(subscriberUserJwt),
+        destinationOne,
+        IsolationStrategy.Tenant_User
+      );
+      const actual = await getDestination({
+        destinationName: destName,
+        isolationStrategy: IsolationStrategy.Tenant_User,
+        jwt: subscriberUserJwt,
+        iasToXsuaaTokenExchange: false
+      });
+      expect(actual).toEqual(destinationOne);
+    });
+
+    it('ignores cache if isolation requires user JWT but the JWT is not provided', async () => {
```

Comment:
maybe adding a case:
'ignores cache if isolation requires tenant info but the tenant info is missing in the jwt'